deploy/helm/stac-scaler/values.yaml

```
@@ -0,0 +1,155 @@
+
+deploymentNamespace: pgstac
+envCode: stac
+
+processors:
+  staccollection:
+    template: processor-stac-collection
+    labels: 
+      app: app-stac-collection
+    image:
+      repository: "" # format: acr.azurecr.io
+      name: stac-cli
+      pullPolicy: Always
+      tag: "latest"
+    env:
+      SERVICE_BUS_CONNECTION_STRING: ""
+      TOPIC_NAME: "staccollection"
+      SUBSCRIPTION_NAME: "staccollectiontopic"
+      DATA_STORAGE_ACCOUNT_CONNECTION_STRING: ""
+      DATA_STORAGE_ACCOUNT_NAME: ""
+      DATA_STORAGE_ACCOUNT_KEY: ""
+      STACCOLLECTION_STORAGE_CONTAINER_NAME: ""
+      AZURE_LOG_CONNECTION_STRING: ""
+      PGHOST: ""
+      PGPORT: "5432"
+      PGUSER: ""
+      PGDATABASE: ""
+      PGPASSWORD: ""
+      WORKING_DIR: "/app"
+      PYTHONUNBUFFERED: 0
+    podAnnotations: {}
+    resources:
+      requests:
+        memory: "718Mi"
+      limits:
+        cpu: "100m"
+        memory: "1024Mi"
+  staceventconsumer:
+    template: processor-stac-event-consumer
+    labels: 
+      app: app-stac-event-consumer
+    image:
+      repository: "" # format: acr.azurecr.io
+      name: stac-cli
+      pullPolicy: Always
+      tag: "latest"
+    env:
+      SERVICE_BUS_CONNECTION_STRING: ""
+      TOPIC_NAME: "pgstactopic"
+      SUBSCRIPTION_NAME: "pgstacsubscription"
+      DATA_STORAGE_ACCOUNT_CONNECTION_STRING: ""
+      GENERATED_STAC_STORAGE_CONTAINER_NAME: ""
+      AZURE_LOG_CONNECTION_STRING: ""
+      DATA_STORAGE_PGSTAC_CONTAINER_NAME: ""
+      PGHOST: ""
+      PGPORT: "5432"
+      PGUSER: ""
+      PGDATABASE: ""
+      PGPASSWORD: ""
+      MESSAGE_COUNT: 50
+      WORKING_DIR: "/app"
+      PYTHONUNBUFFERED: 0
+    podAnnotations: {}
+    resources:
+      requests:
+        memory: "1024Mi"
+      limits:
+        cpu: "100m"
+        memory: "1536Mi"
+  generatestacjson:
+    template: processor-generate-stac-json
+    labels:
+      app: app-generate-stac-json
+    image:
+      repository: "" # format: acr.azurecr.io
+      name: stac-cli
+      pullPolicy: Always
+      tag: "latest"
+    env:
+      DATA_STORAGE_ACCOUNT_CONNECTION_STRING: ""
+      DATA_STORAGE_ACCOUNT_NAME: ""
+      DATA_STORAGE_ACCOUNT_KEY: ""
+      STACIFY_STORAGE_CONTAINER_NAME: ""
+      SERVICE_BUS_CONNECTION_STRING: ""
+      TOPIC_NAME: "stacify"
+      SUBSCRIPTION_NAME: "stacifytopic"
+      GENERATED_STAC_STORAGE_CONTAINER_NAME: ""
+      AZURE_LOG_CONNECTION_STRING: ""
+      DATA_STORAGE_PGSTAC_CONTAINER_NAME: ""
+      STAC_METADATA_TYPE_NAME: ""
+      JPG_EXTENSION: "200.jpg"
+      XML_EXTENSION: "aux.xml"
+      COLLECTION_ID: "naip"
+      MESSAGE_COUNT: 50
+      WORKING_DIR: "/app"
+      PYTHONUNBUFFERED: 0
+    podAnnotations: {}
+    resources:
+      requests:
+        memory: "1024Mi"
+      limits:
+        cpu: "100m"
+        memory: "1536Mi"
+
+stacfastapi:
+  replicas: 1
+  privateIp: 10.6.3.15
+  labels:
+    app: app-stacfastapi
+  image:
+    repository: "" # format: acr.azurecr.io
```

I still think this could lead to confusion, since the default is an empty string. This value, for example is used here:
`image: "{{ .Values.stacfastapi.image.repository }}/{{ .Values.stacfastapi.image.name }}:{{ .Values.stacfastapi.image.tag }}"`
If the templated is deployed without this value modified, then the resulting `image` value will be `/stac-fastapi:latest` which is invalid. Instead, if this is changed to
```suggestion
    #repository: myacr.azurecr.io
```
and the declaration in the deployment YAML file is changed to something like:
```yaml
image: "{{ required "The image repository must be specified" .Values.stacfastapi.image.repository }}/{{ .Values.stacfastapi.image.name }}:{{ .Values.stacfastapi.image.tag }}"
```
Then you'll get a failure rendering the helm template instead of an invalid k8s deployment. The next step would be to make global defaults for the image repo and tag, since those are likely to be customized globally anyway. This is best put in a template in `_helpers.yaml`. See the [Apache Airflow chart](https://github.com/apache/airflow/blob/main/chart/templates/_helpers.yaml#L251) for one example of this.

---

Implemented as suggested.